### PR TITLE
ENG-2175 Pages management tests added and other improvements

### PR DESCRIPTION
**Pull request by alepintus**

### CODE REVIEW

**Comment by entando-jx** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **alepintus**

If they are not already assigned, you can assign the PR to them by writing `/assign @alepintus` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=entando%2Fapp-builder).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/entando/app-builder/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: test/cypress/support/command/command.js [hunk 1]
```diff
@@ -12,6 +14,24 @@ Cypress.Commands.add('getByTestId', (selector, ...args) => cy.get(`[${TEST_ID_KE
...
+ *  Get input elements by name
+ *  @param selector - The name of the DOM elements
+ */
+Cypress.Commands.add('getInputByName', (name, ...args) => cy.get(`input[name=${escapeRegExp(name)}]`, ...args));
+
+/**
+ *  Get Typeahead Option by aria label
+ *  @param value - The value of the aria label
+ */
+Cypress.Commands.add('getTypeaheadOption', (value, ...args) => cy.get('.dropdown-item', ...args).contains(value));
```
Comment (gidesan) [inline test/cypress/support/command/command.js:27]: seems that it's not using ARIA label

File: test/cypress/support/command/page-command.js [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+  });
+  cy.getModalDialogByTitle('Delete page').should('be.visible');
+  cy.getButtonByText('Delete').click();
+});
+
+/**
+ * Publish a Page from the page tree
+ * @param pageCode - the page code to publish
+ */
+Cypress.Commands.add('publishPageAction', (pageCode) => {
```
Comment (gidesan) [inline test/cypress/support/command/page-command.js:127]: is the `action` suffix really needed? do we need to know the page was published by clicking on the kebab action or elsewhere?

Comment (alepintus) [reply test/cypress/support/command/page-command.js:127]: Yes we have another publish page command in page designer (publishPageClick) and I prefer to differ this from the other (action refers to the actions menu)

File: test/cypress/support/command/command.js (hunk 1 above)
Comment (alepintus) [reply test/cypress/support/command/command.js:27]: nope, this is the only way I found to access properly to the options without changing the typeahead component

Comment (gidesan) [reply test/cypress/support/command/command.js:27]: then please amend ```*  Get Typeahead Option by aria label
 *  @param value - The value of the aria label```

Comment (alepintus) [reply test/cypress/support/command/command.js:27]: Done

File: test/cypress/support/command/command.js [hunk 3]
```diff
@@ -12,6 +14,24 @@ Cypress.Commands.add('getByTestId', (selector, ...args) => cy.get(`[${TEST_ID_KE
...
+ *  Get input elements by name
+ *  @param selector - The name of the DOM elements
+ */
+Cypress.Commands.add('getInputByName', (name, ...args) => cy.get(`input[name=${escapeRegExp(name)}]`, ...args));
+
+/**
+ *  Get Typeahead Option by value
+ *  @param value - The the exact Option text displayed in the UI
+ */
+Cypress.Commands.add('getTypeaheadOption', (value, ...args) => cy.get('.dropdown-item', ...args).contains(value));
```
Comment (gidesan) [inline test/cypress/support/command/command.js:27]: then `getTypeaheadOptionByValue` is probably clearer

**Review by gidesan — CHANGES_REQUESTED** [review]

File: test/cypress/support/command/command.js (hunk 3 above)
Comment (alepintus) [reply test/cypress/support/command/command.js]: Done

**Review by gidesan — CHANGES_REQUESTED** [review]
Almost done :)

File: test/cypress/support/command/page-management-command.js [hunk 4]
```diff
@@ -56,15 +55,15 @@ function expandOrCollapse(folderName, isExpand) {
  * Expand one folder by click on the arrow aside the name.
  * @param folderName the exact name displayed in the UI e.g. "Home", "Sitemap"
  */
-Cypress.Commands.add('expand', (folderName) => {
+Cypress.Commands.add('expandPageTree', (folderName) => {
```
[the remark this answers is not in the document]
Comment (alepintus) [reply test/cypress/support/command/page-management-command.js]: Commands changed as suggested

File: test/cypress/support/command/page-management-command.js [hunk 5]
```diff
@@ -75,14 +74,14 @@ function expandOrCollapseAll(label) {
 /**
  * Expand all folders by clicking on the "Expand" main button above the page tree.
  */
-Cypress.Commands.add('expandAll', () => {
+Cypress.Commands.add('expandAllPageTree', () => {
```
[the remark this answers is not in the document]
Comment (alepintus) [reply test/cypress/support/command/page-management-command.js]: Commands changed as suggested

**Review by gidesan — APPROVED** [review]

**Comment by entando-jx** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/entando/app-builder/pull/1052#pullrequestreview-640788122" title="Approved">gidesan</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=entando%2Fapp-builder).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/entando/app-builder/blob/master/OWNERS)~~ [gidesan]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**